modules/core/src/main/java/org/apache/ignite/spi/discovery/tcp/ServerImpl.java

```diff
@@ -7287,10 +7288,21 @@ private InetSocketAddress checkConnection(List<InetSocketAddress> addrs, int tim
                                     sock.connect(addr, perAddrTimeout);
 
                                     liveAddrHolder.compareAndSet(null, addr);
+
+                                    if (log.isInfoEnabled()) {
```

Comment:
Can we begin forming long string somewhere above like: `String logMsg = "Connection check to node: [nodeId=" + node.id() + ", addressToCheck=" + addr + ", result="`, add actual result/error later and log once in the final block?

Reply:
I've tried to refactor it with StringBuilder. It's not really readable. I think it's going to be harder to maintain. I've commited the update. Take a look, what do you think?